### PR TITLE
1219677 test predictive keyboard gij

### PR DESCRIPTION
**Pull request by raylin**

### CODE REVIEW

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=539505507e1f86e34d02a4d52b3c9bf7b25ba102

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=348c47e8495e31e3a88cf7c647dc39c4c9b30255

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=071161f694b6047f7f6a335750982344675486f9

File: apps/keyboard/test/marionette/auto_correction_test.js [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+  setup(function() {
+    keyboard =  new Keyboard(client);
+    keyboardTestApp = new KeyboardTestApp(client);
+    settings = new Settings(client);
+  });
+
+  suite('autocorrect toggle', function() {
+    setup(function() {
+      keyboard.launch();
+      keyboard.switchToBuiltInSettings();
```
Comment (timdream) [inline apps/keyboard/test/marionette/auto_correction_test.js]: `switchToSettings`?

File: apps/keyboard/test/marionette/auto_correction_test.js [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+      // Toggle one time
+      keyboard.clickAutocorrectOption();
+      assert.equal(keyboard.autocorrect, !currentSwitch);
+
+      // Toggle again
+      keyboard.clickAutocorrectOption();
+      assert.equal(keyboard.autocorrect, !!currentSwitch);
+    });
+
+    test('re-open built-in settings', function() {
```
Comment (timdream) [inline apps/keyboard/test/marionette/auto_correction_test.js]: `built-in keyboard`

File: apps/keyboard/test/marionette/auto_correction_test.js [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+    test('bug 1073870', function() {
+      keyboard.type('Hello worl');
+      keyboardTestApp.switchTo();
+
+      client.executeScript(
+        'var el = document.activeElement;' +
+        'el.selectionStart = el.selectionEnd = 5;');
+
+      keyboard.switchTo();
+      keyboard.type(' ');
```
Comment (timdream) [inline apps/keyboard/test/marionette/auto_correction_test.js]: You might need to wait for suggestion to disappear before typing .. this is racey.

File: apps/keyboard/test/marionette/auto_correction_test.js [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+      }
+
+      keyboardTestApp.launch();
+      keyboardTestApp.switchTo();
+      keyboardTestApp.textInput.tap();
+
+      keyboard.switchTo();
+    });
+
+    test('no autocorrect while texting', function() {
```
Comment (timdream) [inline apps/keyboard/test/marionette/auto_correction_test.js]: `texting` -> `typing`

File: apps/keyboard/test/marionette/auto_correction_test.js [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+
+      keyboard.switchTo();
+      keyboard.tapBackspaceKey();
+      keyboardTestApp.switchTo();
+
+      assert.equal(
+        keyboardTestApp.textInput.getAttribute('value'), 'keyboard Tea he wot');
+    });
+
+    test('bug 1073870', function() {
```
Comment (timdream) [inline apps/keyboard/test/marionette/auto_correction_test.js]: Let's change the description of this bug to something meaningful and leave the bug # in a comment above.

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5fe1c97b27a01d8bd9da846999263e12fe93c56c

**Pull request merged**